lms/djangoapps/course_home_api/course_metadata/v1/views.py

```diff
@@ -83,7 +83,7 @@ def get(self, request, *args, **kwargs):
         username = request.user.username if request.user.username else None
         course = course_detail(request, request.user.username, course_key)
         enrollment = CourseEnrollment.get_enrollment(request.user, course_key_string)
```

Comment:
i think you can just use the is_enrolled method instead here since that checks the is_active field

Reply:
This was changed in your PR to pass enrollment through the context. https://github.com/edx/edx-platform/pull/27505/files#diff-9e7feaacbd92cb7b7f465c75629aaf34bb5f3002b032dcefdd15168231128731L84-L90

Reply:
right, i was suggesting you could use is_enrolled in addition, potentially if we're worried about extra calls you could just have is_enrolled take the enrollment as an optional kwarg?

Reply:
I'm not sure I understand your concern here Matthew. There is no additional query to look at a field on an already fetched object and we have to fetch this object regardless for the context below. Are you arguing in the sense of clarity of code? I suppose it doesn't matter one way or the other. No additional queries will happen even if switched to `CourseEnrollment.is_enrolled(user, course.id)` because it uses a request cache.

Reply:
no i was just suggesting that if we reuse the is_enrolled method we could reduce the number of places we need to check the is_active logic
it isn't a lot of code so probably not a big deal but seems like it could be nice to consolidate where we make that check